src/app/util/ember-compatibility-functions.cpp

```diff
@@ -936,6 +936,36 @@ CHIP_ERROR WriteSingleClusterData(const SubjectDescriptor & aSubjectDescriptor,
     return apWriteHandler->AddStatus(aPath, status);
 }
 
+bool IsClusterDataVersionAllowed(const EndpointId & aEndpointId, const ClusterId & aClusterId, const DataVersion & aDataVersion)
+{
+    DataVersion * version = emberAfDataVersionStorage(aEndpointId, aClusterId);
+    if (version == nullptr)
+    {
+        ChipLogError(DataManagement, "Endpoint %" PRIx16 ", Cluster " ChipLogFormatMEI " not found in IsClusterDataVersionAllowed!",
+                     aEndpointId, ChipLogValueMEI(aClusterId));
+        return false;
+    }
+    else
+    {
+        return (*(version)) > aDataVersion ? true : false;
```

Comment:
No, this is not the right check.  The only meaningful check on data versions is an equality check.

Why do we think we want to do a `>` here?

Reply:
per offline discussion, equal check is enough for our write

Reply:
updated code